### PR TITLE
[torchlib] Implement `aten::prelu`

### PR DESCRIPTION
**Pull request by justinchuby**

### CODE REVIEW

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 1]
```diff
@@ -6565,10 +6565,20 @@ def aten_pow(self: TReal, exponent: TTensor) -> TReal:
...
-    raise NotImplementedError()
+    zero = op.CastLike(0, self)
+    rank = len(self.shape)
+    if rank == 0:
+        if len(weight.shape) == 1:
+            # e.g. [] * [1]
+            weight = op.Squeeze(weight, [-1])
+    elif rank >= 2:
+        # e.g. [5,10,5] * [10]
+        weight = op.Reshape(weight, [1, -1] + [1] * (rank - 2))
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:6580]:
```suggestion
        if len(weight.shape) == 1:
            # e.g. self: [], weight: [1]
            weight = op.Squeeze(weight, [-1])
    elif rank >= 2:
        # e.g. self: [5,10,5], weight: [10]
        weight = op.Reshape(weight, [1, -1] + [1] * (rank - 2))
```

**Review by titaiwangms — APPROVED** [review]

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 2]
```diff
@@ -6565,10 +6565,20 @@ def aten_pow(self: TReal, exponent: TTensor) -> TReal:
...
+def aten_prelu(self: TReal, weight: TReal) -> TReal:
     """prelu(Tensor self, Tensor weight) -> Tensor"""
 
-    raise NotImplementedError()
+    zero = op.CastLike(0, self)
+    rank = len(self.shape)
+    if rank == 0:
+        if len(weight.shape) == 1:
+            # e.g. self: [], weight: [1]
+            weight = op.Squeeze(weight, [-1])
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:6577]:
```suggestion
        # e.g. self: [], weight: [1]
        weight = op.Squeeze(weight, [-1])
```

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 3]
```diff
@@ -6565,10 +6565,19 @@ def aten_pow(self: TReal, exponent: TTensor) -> TReal:
...
+@torch_op(("aten::prelu", "aten::_prelu_kernel"), trace_only=True)
+def aten_prelu(self: TReal, weight: TReal) -> TReal:
     """prelu(Tensor self, Tensor weight) -> Tensor"""
 
-    raise NotImplementedError()
+    zero = op.CastLike(0, self)
+    rank = len(self.shape)
+    if rank == 0:
+        # e.g. self: [], weight: [1]
+        weight = op.Squeeze(weight, [-1])
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:6576]:
```suggestion
        weight = op.Squeeze(weight)
```

**Comment by titaiwangms** [comment]
Is there a reason we don't use ONNX PRelu? https://github.com/onnx/onnx/blob/main/docs/Operators.md#PRelu

**Comment by justinchuby** [comment]
I didn't know it existed. I will update.

**Pull request merged**